dev_docs/tutorials/saved_objects.mdx

```diff
@@ -0,0 +1,260 @@
+
+export const dashboardVisualization: SavedObjectsType = {
+  name: 'dashboard_visualization', [1]
+  hidden: false,
+  namespaceType: 'single',
+  mappings: {
+    dynamic: false,
+    properties: {
+      description: {
+        type: 'text',
+      },
+      hits: {
+        type: 'integer',
+      },
+    },
+  },
+  migrations: {
+    '1.0.0': migratedashboardVisualizationToV1,
+    '2.0.0': migratedashboardVisualizationToV2,
+  },
+};
+```
+
+[1] Since the name of a Saved Object type forms part of the url path for the public Saved Objects HTTP API, 
+these should follow our API URL path convention and always be written as snake case.
+
+**src/plugins/my_plugin/server/saved_objects/index.ts**
+
+```ts
+export { dashboardVisualization } from './dashboard_visualization';
+export { dashboard } from './dashboard';
+```
+
+**src/plugins/my_plugin/server/plugin.ts**
+
+```ts
+import { dashboard, dashboardVisualization } from './saved_objects';
+
+export class MyPlugin implements Plugin {
+  setup({ savedObjects }) {
+    savedObjects.registerType(dashboard);
+    savedObjects.registerType(dashboardVisualization);
+  }
+}
+```
+
+## Mappings
+
+Each Saved Object type can define it’s own Elasticsearch field mappings. Because multiple Saved Object
+types can share the same index, mappings defined by a type will be nested under a top-level field that matches the type name.
+
+For example, the mappings defined by the dashboard_visualization Saved Object type:
+
+**src/plugins/my_plugin/server/saved_objects/dashboard_visualization.ts**
+
+```ts
+import { SavedObjectsType } from 'src/core/server';
+
+export const dashboardVisualization: SavedObjectsType = {
+  name: 'dashboard_visualization',
+  ...
+  mappings: {
+    properties: {
+      dynamic: false,
+      description: {
+        type: 'text',
+      },
+      hits: {
+        type: 'integer',
+      },
+    },
+  },
+  migrations: { ... },
+};
+```
+
+Will result in the following mappings being applied to the .kibana index:
+
+```ts
+{
+  "mappings": {
+    "dynamic": "strict",
+    "properties": {
+      ...
+      "dashboard_vizualization": {
+        "dynamic": false,
+        "properties": {
+          "description": {
+            "type": "text",
+          },
+          "hits": {
+            "type": "integer",
+          },
+        },
+      }
+    }
+  }
+}
+```
+Do not use field mappings like you would use data types for the columns of a SQL database. Instead, field mappings are analogous to a 
+SQL index. Only specify field mappings for the fields you wish to search on or query. By specifying dynamic: false
+ in any level of your mappings, Elasticsearch will accept and store any other fields even if they are not specified in your mappings.
+
+Since Elasticsearch has a default limit of 1000 fields per index, plugins should carefully consider the 
+fields they add to the mappings. Similarly, Saved Object types should never use dynamic: true as this can cause an arbitrary
```

Comment:
nit
```suggestion
fields they add to the mappings. Similarly, Saved Object types should never use `dynamic: true` as this can cause an arbitrary
```